### PR TITLE
[feature](merge-cloud) add available cloud cluster for load and query

### PR DESCRIPTION
**Pull request by luwei16**

### CODE REVIEW

**Comment by luwei16** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java [hunk 1]
```diff
@@ -193,6 +193,23 @@ private void onPendingTaskFinished(BrokerPendingTaskAttachment attachment) {
         loadStartTimestamp = System.currentTimeMillis();
     }
 
+    private LoadLoadingTask createAndInit(Database db, OlapTable table, List<BrokerFileGroup> brokerFileGroups,
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java:196]: just call it createLoadingTask()

File: fe/fe-core/src/main/java/org/apache/doris/cloud/planner/CloudStreamLoadPlanner.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    private AutoCloseConnectContext buildConnectContext() throws UserException {
+        if (ConnectContext.get() == null) {
+            ConnectContext connectContext = new ConnectContext();
+            connectContext.setCloudCluster();
+            if (Strings.isNullOrEmpty(cloudClusterName)) {
+                connectContext.setCloudCluster();
+            } else {
+                connectContext.setCloudCluster(cloudClusterName);
+            }
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/planner/CloudStreamLoadPlanner.java:52]: whats the difference between setCloudCluster() and setCloudCluster(cluster)
can we merge them into one method?
if not, changing the former one to another name like chooseCloudCluster() may avoid the confusion

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 3]
```diff
@@ -1044,6 +1047,33 @@ public String getCloudCluster() {
         return cloudCluster;
     }
 
+    public void setCloudCluster() {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:1050]: pls add some commment here to desc what is going on

File: fe/fe-core/src/main/java/org/apache/doris/cloud/load/loadv2/CloudBrokerLoadJob.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import java.util.UUID;
+
+public class CloudBrokerLoadJob extends BrokerLoadJob {
+    private static final Logger LOG = LogManager.getLogger(CloudBrokerLoadJob.class);
+
+    protected static final String CLUSTER_ID = "clusterId";
+    protected String clusterId;
+
+    public CloudBrokerLoadJob() {
+        super();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/load/loadv2/CloudBrokerLoadJob.java:51]: this call is redundant

File: fe/fe-core/src/main/java/org/apache/doris/cloud/load/loadv2/CloudBrokerLoadJob.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.UUID;
+
+public class CloudBrokerLoadJob extends BrokerLoadJob {
+    private static final Logger LOG = LogManager.getLogger(CloudBrokerLoadJob.class);
+
+    protected static final String CLUSTER_ID = "clusterId";
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/load/loadv2/CloudBrokerLoadJob.java:47]: it seems to be confusing with the existing clusterid stored on BE.
is it better using cloudClusterId instead?

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Comment by gavinchou** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**